vertx-web/src/main/java/io/vertx/ext/web/handler/impl/CorsHandlerImpl.java

```diff
@@ -167,13 +165,22 @@ private void sendInvalid(HttpServerResponse resp) {
   }
 
   private boolean isValidOrigin(String origin) {
-    if (allowedOrigin == null) {
+    if (allowedOrigin == null && isValidOriginURI(origin)) {
       // Null means accept all origins
       return true;
     }
     return allowedOrigin.matcher(origin).matches();
   }
 
+  private boolean isValidOriginURI(String origin) {
```

Comment:
Is this really necessary? we're paying a parsing tax while later we will match it against a regular expression, I'd assume that the regex would catch errors.

Reply:
This is useful to check if the Origin header contains a valid value when the allowed origin is a wildcard https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin#Syntax